packages/core-transaction-pool/src/query.ts

```diff
@@ -0,0 +1,156 @@
+import { Container, Contracts } from "@arkecosystem/core-kernel";
+import { Interfaces, Transactions } from "@arkecosystem/crypto";
+
+import { IteratorMany } from "./utils";
+
+class QueryIterator implements Iterator<Interfaces.ITransaction> {
+    private readonly iter: Iterator<Interfaces.ITransaction>;
+    private readonly predicate: Contracts.TransactionPool.Predicate | undefined;
+
+    public constructor(iter: Iterator<Interfaces.ITransaction>, predicate?: Contracts.TransactionPool.Predicate) {
+        this.iter = iter;
+        this.predicate = predicate;
+    }
+
+    public next(value?: any): IteratorResult<Interfaces.ITransaction> {
+        for (let r = this.iter.next(); !r.done; r = this.iter.next()) {
```

Comment:
Avoid single character variable assignments unless it is a `for` loop with numerical increments and also complex loop conditions like this as they are harder to read and make sense of than a `while` or `for (const x of ...)`.

Could this be replaced with a `while` that loops until it has a match or `next()` no longer performs anything?

Reply:
Replaced it with while loop

Reply:
Replaced it with generator and for..of loop